Sources/Core/UInt32.swift

```diff
@@ -80,6 +80,28 @@ extension UInt32: Numeric {
   }
 }
 
+extension UInt32: BinaryInteger {
+  @_transparent
+  public static func /(lhs: UInt32, rhs: UInt32) -> UInt32 {
+    precondition(rhs != (0 as UInt32), "Division by zero")
+
+    let (result, overflow) =
+        (Builtin.udiv_Int32(lhs._value, rhs._value), false._value)
+    Builtin.condfail_message(overflow,
+                             StaticString("arithmetic overflow")
+                                .unsafeRawPointer)
+    return UInt32(result)
+  }  
+  
+  @_transparent
+  public static func %(lhs: UInt32, rhs: UInt32) -> UInt32 {
+    precondition(rhs == (0 as UInt32), 
```

Comment:
I think you meant `!=`.

```suggestion
    precondition(rhs != (0 as UInt32), 
```